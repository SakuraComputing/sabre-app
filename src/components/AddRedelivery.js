import React from 'react';
import { connect } from 'react-redux';
import RedeliveryForm from './RedeliveryForm';
import { createRedelivery } from "../actions/redelivery";
import UrlConstants from '../../config/url-constants';

export class AddRedelivery extends React.Component {
    onSubmit = (redelivery) => {
      this.props.createRedelivery(redelivery);
      this.props.history.push(UrlConstants.REDELIVERY_HOME);
    };
    render() {
        return (
            <div>
                <h2>Add Redelivery</h2>
                <RedeliveryForm
                    onSubmit = {this.onSubmit}
                />
            </div>
        );
    }
}

export const mapDispatchToProps = (dispatch) => ({
    createRedelivery: (redelivery) => dispatch(createRedelivery(redelivery))
});
export default connect(undefined, mapDispatchToProps)(AddRedelivery);