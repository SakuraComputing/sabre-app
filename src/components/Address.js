import React from 'react';

const Address = ( { address } ) => (
    <table className='address-panel'>
        <tbody>
            <tr>{address.addressName}</tr>
            <tr>{address.addressLine1}</tr>
            <tr>{address.addressLine2}</tr>
            <tr>{address.town}</tr>
            <tr>{address.county}</tr>
            <tr>{address.telephone}</tr>
        </tbody>
    </table>
);
export default Address;
