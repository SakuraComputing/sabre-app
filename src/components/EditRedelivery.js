import React from 'react';
import { connect } from 'react-redux';
import { updateRedelivery, deleteRedelivery} from '../actions/redelivery';
import RedeliveryForm from './RedeliveryForm';
import UrlConstants from "../../config/url-constants";

const REDELIVERY_URL = UrlConstants.REDELIVERY_ID;

export class EditRedelivery extends React.Component {

    onSubmit = (redelivery) => {
        this.props.updateRedelivery(this.props.redelivery.id, redelivery);
        this.props.history.push('/redelivery');
    };

    onDelete = (id) => {
        this.props.deleteRedelivery( this.props.cardId);
        this.props.history.push('/redelivery');
    };

    render() {
        return (
            <div>
                <h2>Edit Redelivery</h2>
                <RedeliveryForm
                    redelivery={this.props.redelivery}
                    onSubmit={this.onSubmit}
                />
                <button
                    onClick={this.onDelete}>
                    Delete Redelivery
                </button>
            </div>
        );
    }
}

export const mapStateToProps = ( state, props ) => ({
   redelivery: state.redeliveries.find((redelivery) => {
       return redelivery.cardId === props.match.params.id;
   })
});

export const mapDispatchToProps = (dispatch) => ({
    updateRedelivery: (id, redelivery) => dispatch(updateRedelivery(id, redelivery)),
    deleteRedelivery: (data) => dispatch(deleteRedelivery(data))
});
export default connect(mapStateToProps, mapDispatchToProps)(EditRedelivery);
