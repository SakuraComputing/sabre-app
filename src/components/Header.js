import React from 'react';
import { NavLink } from 'react-router-dom';
import { Navbar } from 'react-bootstrap';

const Header = () => (
    <div className='menu-container'>
        <Navbar className="navbar navbar-default navbar-static-top" >
            <ul>

                <li>
                    <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
                </li>
                <li>
                    <NavLink to="/tracking" activeClassName="is-active">Parcel Tracking</NavLink>
                </li>
                <li>
                    <NavLink to="/redelivery" activeClassName="is-active">Redelivery</NavLink>
                </li>
            </ul>
        </Navbar>
        <h1>Yodel Web Portal</h1>
    </div>
);
export default Header;
