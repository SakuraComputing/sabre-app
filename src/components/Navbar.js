import React from 'react';

export default class Navbar extends React.Component {

    constructor(props) {
        super(props);
    }

    onHamburgerClick = () =>  {
      this.props.onHamburgerClick && this.props.onHamburgerClick();
    };

    render(){
        return (
            <nav className="navbar navbar-default navbar-fixed-top">
                <div className="container-fluid">
                    <div className="navbar-header">
                        <button type="button" className="navbar-btn navbar-toggle collapsed" onClick={this.onHamburgerClick}>
                            <span className="sr-only">Toggle Navigation</span>
                            <span className="icon-bar"></span>
                            <span className="icon-bar"></span>
                            <span className="icon-bar"></span>
                        </button>
                    </div>
                    {this.props.children}
                </div>
            </nav>
        );
    }
}