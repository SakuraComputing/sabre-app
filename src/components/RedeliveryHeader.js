import React from 'react';
import { connect } from 'react-redux';
import {setCardId, setParcelId, setReqStartDate, setReqEndDate} from '../actions/filters';
import {setUpRedeliveryData} from '../actions/redelivery';
import UrlConstants from '../../config/url-constants';

export class RedeliveryHeader extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            cardId: '',
            parcelId: '',
            latestScan: undefined,
            requestDate: undefined
        }
    }

    onCardIdChange = (e) => {
        this.props.setCardId(e.target.value);
    };

    onParcelIdChange = (e) => {
        this.props.setParcelId(e.target.value);
    };

    onSetReqStartDate = (e) => {
        this.props.setReqStartDate((e.target.value));
    };

    onSetReqEndDate = (e) => {
        this.props.setReqEndDate((e.target.value));
    };

    getRedeliveries = () => {
        fetch(UrlConstants.REDELIVERIES_URL)
            .then(data => data.json())
            .then(data => {
                this.props.setRedeliveryData(data)
            });
    };

    componentDidMount() {
        this.getRedeliveries();
    }

    render() {
        return (

            <div>
                <form className="select">
                    <p></p>
                    <p></p>
                    <label>Card Id</label>
                    <input
                           id="cardId"
                           type="text"
                           className="card-id"
                           value={this.props.filters.cardId}
                           onChange={this.onCardIdChange}
                           placeholder="Card Id"
                    />
                    Parcel Id:
                    <input
                        type="text"
                        className='parcel-id'
                        value={(this.props.filters.parcelId)}
                        onChange={this.onParcelIdChange}
                        placeholder="Parcel Id"
                    />
                    Request Date:
                    <input
                        className={'request-start-date'}
                        type="date"
                        value={(this.props.filters.startDate)}
                        onChange={this.onSetReqStartDate}
                        placeholder="Request Date"
                    />
                    <input
                        className={'request-end-date'}
                        type="date"
                        value={(this.props.filters.endDate)}
                        onChange={this.onSetReqEndDate}
                    />
                </form>
                <p></p>
            </div>
        )
    }
}
export const mapStateToProps = (state) => ({
    filters: state.filters
});

export const mapDispatchToProps = (dispatch) => ({
    setCardId: (cardId) => dispatch(setCardId(cardId)),
    setParcelId: (parcelId) => dispatch(setParcelId(parcelId)),
    setReqStartDate: (startDate) => dispatch(setReqStartDate(startDate)),
    setReqEndDate: (endDate) => dispatch(setReqEndDate(endDate)),
    setRedeliveryData: (redeliveries) => dispatch(setUpRedeliveryData(redeliveries))
});

export default connect(mapStateToProps, mapDispatchToProps)(RedeliveryHeader);