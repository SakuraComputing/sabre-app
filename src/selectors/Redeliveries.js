import moment from 'moment';

export default (redeliveries, { cardId, parcelId, startDate, endDate }) => {
    return redeliveries.filter((redelivery) => {
        const cardIdMatch = cardId ? redelivery.cardId.toLowerCase().includes(cardId.toLowerCase()) : true;
        const parcelIdMatch = parcelId ? redelivery.parcelId.toLowerCase().includes(parcelId.toLowerCase()) : true;
        const createdMoment = moment(redelivery.requestDate);
        const startDateMatch = startDate ? moment(startDate).isSameOrBefore(createdMoment, 'day') : true;
        const endDateMatch = endDate ? moment(endDate).isSameOrAfter(createdMoment, 'day') : true;
        return cardIdMatch && parcelIdMatch && startDateMatch && endDateMatch;
    });
};