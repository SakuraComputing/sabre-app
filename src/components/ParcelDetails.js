import React from 'react';
import Address from "./Address";

const ParcelDetails = ({ parcel, address }) => (
    <table className='parcel-details' frameBorder="1">
        <tbody>
            <tr>Address:</tr>
            <tr className="address-details" >{address ? <Address address={address}/> : 'Address not found'}</tr>
            <tr>Service Code: {parcel.serviceCode}</tr>
        </tbody>
    </table>
);
export default ParcelDetails;
