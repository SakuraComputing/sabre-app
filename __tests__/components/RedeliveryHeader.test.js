import React from 'react';
import { shallow } from 'enzyme';
import { RedeliveryHeader, mapStateToProps, mapDispatchToProps } from '../../src/components/RedeliveryHeader';
import Redeliveries from '../fixtures/redeliveries';
import moment from 'moment';
// require('es6-promise').polyfill();
// require('isomorphic-fetch');


let wrapper, setCardId, setParcelId, setRedeliveryData, setReqStartDate, setReqEndDate, fetch;

beforeEach(() => {
    const filters = {
        cardId: '',
        parcelId: '',
        latestScan: 0,
        requestDate: 0
    };
    setCardId = jest.fn();
    setRedeliveryData = jest.fn();
    setParcelId = jest.fn();
    setReqStartDate = jest.fn();
    setReqEndDate = jest.fn();
    wrapper = shallow(
           <RedeliveryHeader
                setCardId={setCardId}
                setParcelId={setParcelId}
                setRedeliveryData={setRedeliveryData}
                setReqStartDate={setReqStartDate}
                setReqEndDate={setReqEndDate}
                filters={filters}
            />
    );
});

describe('Redelivery Header',() => {
    it('should render the form correctly',() => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should handle on card id change',() => {
        const value = '99999';
        wrapper.find('#cardId').simulate('change', {
            target: { value }
        });
        expect(setCardId).toHaveBeenLastCalledWith(value);
    });
    it('should handle parcel Id change',() => {
       const value = 'JD0002';
       wrapper.find('.parcel-id').simulate('change', {
           target: {value}
       });
       expect(setParcelId).toHaveBeenCalledWith(value);
    });
    it('should handle on request start date',() => {
       const value = '01/01/11';
       wrapper.find('.request-start-date').simulate('change', {
           target: { value }
       });
       expect(setReqStartDate).toHaveBeenLastCalledWith(value);
    });
    it('should handle on request end date',() => {
        const value = '01/01/11';
        wrapper.find('.request-end-date').simulate('change', {
            target: { value }
        });
        expect(setReqEndDate).toHaveBeenLastCalledWith(value);
    });
    it('maps state to props',() => {
        const currentState = {
            cardId: '',
            parcelId: '',
            latestScan: 0,
            requestDate: 0
        };
        let state = {
            filters: currentState,
        };
        expect(mapStateToProps(state).filters).toEqual(state.filters);
    });

    describe('call to the fetch api',() => {
       it('should send a GET request', () => {

       });
    });
    describe('maps dispatch to props',() => {
        const mockDispatch = jest.fn();
        const actionProps = mapDispatchToProps(mockDispatch);

        it('should dispatch card id to props',() => {
            const cardId = '99999';
            actionProps.setCardId(cardId);
            expect(mockDispatch).toHaveBeenCalledWith({ cardId: cardId, type: 'SET_CARD_ID' });
        });
        it('should dispatch parcel id to props',() => {
            const parcelId = 'JD0002';
            actionProps.setParcelId(parcelId);
            expect(mockDispatch).toHaveBeenCalledWith({ parcelId: parcelId, type: 'SET_PARCEL_ID' });
        });
        it('should dispatch redelivery data to props',() => {
            const redeliveries = Redeliveries;
            actionProps.setRedeliveryData(Redeliveries);
            expect(mockDispatch).toHaveBeenCalledWith({ redeliveries: Redeliveries, type: 'SET_REDELIVERY_DATA'});
        });
        it('should dispatch start date to props',() => {
            const startDate = moment(0);
            actionProps.setReqStartDate(startDate);
            expect(mockDispatch).toHaveBeenCalledWith({ startDate: startDate, type: 'SET_REQ_START_DATE'});
        });
        it('should dispatch end date to props',() => {
            const endDate = moment(0);
            actionProps.setReqEndDate(endDate);
            expect(mockDispatch).toHaveBeenCalledWith({ endDate: endDate, type: 'SET_REQ_END_DATE'});
        });
    });
});

