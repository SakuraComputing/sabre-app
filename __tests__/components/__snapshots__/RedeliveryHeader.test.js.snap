// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redelivery Header should render the form correctly 1`] = `
<div>
  <form
    className="select"
  >
    <p />
    <p />
    <label>
      Card Id
    </label>
    <input
      className="card-id"
      id="cardId"
      onChange={[Function]}
      placeholder="Card Id"
      type="text"
      value=""
    />
    Parcel Id:
    <input
      className="parcel-id"
      onChange={[Function]}
      placeholder="Parcel Id"
      type="text"
      value=""
    />
    Request Date:
    <input
      className="request-start-date"
      onChange={[Function]}
      placeholder="Request Date"
      type="date"
    />
    <input
      className="request-end-date"
      onChange={[Function]}
      type="date"
    />
  </form>
  <p />
</div>
`;
